### PR TITLE
[Coffee] Rewrite

### PR DESCRIPTION
**Pull request by pernielsentikaer**
- Removed almost all AppleScript
- Updated API
- Added Menubar + subtitle status
- Added preferences to set what you want to have enabled when caffeinating

### CODE REVIEW

File: extensions/coffee/src/decaffeinate.ts [hunk 1]
```diff
@@ -1,14 +1,5 @@
...
-const Decaffeinate = async () => {
-  try {
-    await runAppleScript(`do shell script "pgrep caffeinate"`);
-    await runAppleScript('do shell script "killall caffeinate"');
-    await showHUD("Your Mac is decaffeinated");
-  } catch (_) {
-    await showHUD("Your Mac is already decaffeinated");
-  }
+export default async () => {
+  await stopCaffeinate(true, "Your Mac is now deffeinated");
```
Comment (bibixx) [inline extensions/coffee/src/decaffeinate.ts:4]: Small typo here
```suggestion
  await stopCaffeinate(true, "Your Mac is now decaffeinated");
```

File: extensions/coffee/src/caffeinateFor.tsx [hunk 2]
```diff
@@ -1,47 +1,66 @@
...
+        }
+      },
+      minutes: (value) => {
+        if (value && isNaN(parseInt(value, 10))) {
+          return "minutes must be a number";
+        }
+      },
+      seconds: (value) => {
+        if (value && isNaN(parseInt(value, 10))) {
+          return "seconds must be a number";
```
Comment (bibixx) [inline extensions/coffee/src/caffeinateFor.tsx:47]: For consistency sake let's make those uppercase as well 😄 

```suggestion
          return "Minutes must be a number";
        }
      },
      seconds: (value) => {
        if (value && isNaN(parseInt(value, 10))) {
          return "Seconds must be a number";
```

**Review by bibixx — COMMENTED** [review]
Great job! Found 3 small typos/case inconsistencies 😉

**Review by bibixx — APPROVED** [review]
Looks great to me now 🙂

**Review by mooxl — APPROVED** [review]
Thanks for the rewrite! I've reviewed the changes and I don't have any objections. This looks like a great improvement for the extension:)

**Comment by pernielsentikaer** [comment]
Let's do it! I'm also happy to get it out with the new menubar support

**Pull request merged**